Automattic/vip
# chore(deps): update update-notifier to 7.0.0

## Description

This PR updates `update-notifier` to 7.0.0. Because `update-notifier` is now an ESM, this required several additional steps.

1. `update-notifier` is imported via `await import` - that's the only way to import ESM from CJS
2. Because `babel` tries to replace `await import` with `Promise.resolve().then(() => _interopRequireWildcard(require('update-notifier'))`, I had to disable the `@babel/plugin-proposal-dynamic-import` plugin.
3. This broke `proxy-agent` tests because `jest` did not like `await import` and insisted on the ESM-enabled environment. Luckily, the test works without dynamic imports.
4. I chose to disable `update-notifier` when `NODE_ENV=test` because I did not want to install `cross-env` and pass `NODE_OPTIONS=--experimental-vm-modules` to keep Jest happy. However, we may need to do this someday.
5. One day, we will need to switch to ESM. However, I don't feel adventurous enough to do that now :-)

As a bonus, we have four less medium security vulnerabilities.

## Steps to Test

1. CI should pass - my changes should not break the existing tests
2. `npm run build; ./dist/bin/vip.js` should work (it will invoke `update-notifier`; there should be no errors coming from Node.js). E2E tests will also catch an error here.
